tools/tpm2_ptool.py

```diff
@@ -1063,175 +1291,90 @@ def generate_options(self, group_parser):
             '--userpin',
             help='The User pin.\n'),
 
-    def __call__(self, args):
-
+    # Imports a new key
+    def new_key_create(self, args, sobjctx, sobjauth, objauth, tpm2):
+        privkey = args['privkey']
         path = args['path']
+        # create a tertiary object under the loaded secondary object
+        # Map onto the defaults in tpm2_create and store the attributes into
+        # the db.
+        alg = args['algorithm']
+        if alg !='rsa':
+            sys.exit('Unknown algorithm or algorithm not supported, got "%s"' % alg)
 
-        label = args['label']
-        keylabel= args['key_label']
-
-        id = args['id']
-
-        with Db(path) as db:
-
-            token = db.gettoken(label)
-
-            token_config =  dict_from_kvp(token['config'])
-
-            pobj = db.getprimary(token['pid'])
-
-            with TemporaryDirectory() as d:
-                tpm2 = Tpm2(d, path)
-
-                # Get the primary object encrypted auth value and sokey information
-                # to decode it. Based on the incoming pin
-                is_so = args['sopin'] != None
-                if is_so:
-                    pin = args['sopin']
-                    pinpobjauthkeysalt = token['sopobjauthkeysalt']
-                    pinpobjauthkeyiters = token['sopobjauthkeyiters']
-                    pinpobjauth = token['sopobjauth']
-                else:
-                    pin = args['userpin']
-                    pinpobjauthkeysalt = token['userpobjauthkeysalt']
-                    pinpobjauthkeyiters = token['userpobjauthkeyiters']
-                    pinpobjauth = token['userpobjauth']
-
-                pinpobjauthkeysalt = binascii.unhexlify(pinpobjauthkeysalt)
-                pinpobjauthkey = hash_pass(pin.encode(), iters=pinpobjauthkeyiters, salt=pinpobjauthkeysalt)
-
-                pinpobjauth = AESCipher(pinpobjauthkey['rhash']).decrypt(pinpobjauth)
-
-                # At this point we have recovered the ACTUAL auth value for the primary object, so now we
-                # can load up the seal objects
-                sealobject = db.getsealobject(token['id'])
-
-                if is_so:
-                    sealpub = sealobject['sopub']
-                    sealpriv = sealobject['sopriv']
-                    salt = sealobject['soauthsalt']
-                    iters = sealobject['soauthiters']
-                else:
-                    sealpub = sealobject['userpub']
-                    sealpriv = sealobject['userpriv']
-                    salt = sealobject['userauthsalt']
-                    iters = sealobject['userauthiters']
+        with TemporaryDirectory() as d:
+            tpm2 = Tpm2(d, path)
```

Comment:
The caller of `new_key_create()` passes in a tpm2 object, cant you just use it? No need to reseat the object reference with a new one.

Reply:
I agree, it was a oversight on my part